docs/api/hardhat/getting-started.md

```diff
@@ -2,242 +2,299 @@
 
-```solidity
-//SPDX-License-Identifier: Unlicensed
-pragma solidity ^0.8.13;
-
-contract Greeter {
-    string private greeting;
+```text
+WALLET_PRIVATE_KEY=abcdef12345....
+```
 
-    constructor(string memory _greeting) {
-        greeting = _greeting;
-    }
+## Compile and deploy a contract
 
-    function greet() public view returns (string memory) {
-        return greeting;
-    }
+Smart contracts are placed in the `contracts` folder. The template project contains a simple `Greeter.sol` contract and a script to deploy it.
 
-    function setGreeting(string memory _greeting) public {
-        greeting = _greeting;
-    }
-}
+To compile the contract, run:
+```sh
+yarn hardhat compile
+``` 
+You'll see the following output:
+```text
+Compiling 1 Solidity file
+Successfully compiled 1 Solidity file
+✨  Done in 1.09s.
 ```
-
-3. Run `yarn hardhat compile` which uses the `hardhat-zksync-solc` plugin to compile the contract. The `artifacts-zk` and `cache-zk` folders will be created in the root directory (instead of the regular Hardhat's `artifacts` and `cache`).
+The `artifacts-zk` and `cache-zk` folders will be created in the root directory (instead of the regular Hardhat's `artifacts` and `cache`). These folders contain the compilation artifacts (including contract's ABIs) and compiler cache files.
 
 ::: tip
 
-Note that the `artifacts-zk` and `cache-zk` folders contain compilation artifacts and cache, and should not be added to version control, so it's a good practice to include them in the `.gitignore` file of your project.
+Note that the `artifacts-zk` and `cache-zk` folders  are already included in the `.gitignore` file of the project.
 
 :::
 
-4. Create a deployment script called `deploy.ts`, in the `deploy` folder, and copy/paste the following code into it, replacing `<WALLET-PRIVATE-KEY>` with your private key:
+In the `deploy` folder you'll find the `deploy-greeter.ts` script. This script uses the `Deployer` class from the `hardhat-zksync-deploy` package to deploy the `Greeter.sol` contract. 
+
 
 ```typescript
-import { utils, Wallet } from "zksync-web3";
+import { Wallet, utils } from "zksync-web3";
 import * as ethers from "ethers";
 import { HardhatRuntimeEnvironment } from "hardhat/types";
 import { Deployer } from "@matterlabs/hardhat-zksync-deploy";
 
+// load env file
+import dotenv from "dotenv";
+dotenv.config();
+
+// load wallet private key from env file
+const PRIVATE_KEY = process.env.WALLET_PRIVATE_KEY || "";
+
+if (!PRIVATE_KEY)
+  throw "⛔️ Private key not detected! Add it to the .env file!";
+
 // An example of a deploy script that will deploy and call a simple contract.
 export default async function (hre: HardhatRuntimeEnvironment) {
   console.log(`Running deploy script for the Greeter contract`);
 
   // Initialize the wallet.
-  const wallet = new Wallet("<WALLET-PRIVATE-KEY>");
+  const wallet = new Wallet(PRIVATE_KEY);
```

Comment:
```suggestion
  const wallet = new Wallet("<PRIVATE_KEY>");
```

Reply:
We should always do something like this to define a placeholder and that it's a string.

Reply:
it's actually using a variable defined a few lines above in the script, which loads the value from the env file.

Reply:
argh, no context in git ...